### PR TITLE
Feature/unr 1936 move generators to spatialgdkeditor

### PR DESCRIPTION
**Pull request by aleximprobable**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
I've moved GenerateDefaultLaunchConfig and GenerateDefaultWorkerJson from SpatialGDKEditorToolbar to new files in SpatialGDKEditor.

#### Release note
This is an internal change that's not interesting to UnrealGDK customers.

#### Tests
Local deployment should start successfully.

#### Documentation
No doc update.

#### Primary reviewers
@joshuahuburn @improbable-valentyn

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+DEFINE_LOG_CATEGORY(LogSpatialGDKDefaultLaunchConfigGenerator);
+
+#define LOCTEXT_NAMESPACE "SpatialGDKDefaultLaunchConfigGenerator"
+
+using namespace SpatialGDK;
+
+namespace
+{
+bool WriteFlagSection(TSharedRef< TJsonWriter<> > Writer, const FString& Key, const FString& Value)
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp]: Nit (here and in all other places with nested templates):
```suggestion
bool WriteFlagSection(TSharedRef<TJsonWriter<>> Writer, const FString& Key, const FString& Value)
```

Comment (improbable-valentyn) [reply SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp]: I did see that this was like this before, but would still prefer the no-space version.

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+{
+	Writer->WriteObjectStart();
+		Writer->WriteValue(TEXT("name"), Key);
+		Writer->WriteValue(TEXT("value"), Value);
+	Writer->WriteObjectEnd();
+
+	return true;
+}
+
+bool WriteWorkerSection(TSharedRef< TJsonWriter<> > Writer, const FWorkerTypeLaunchSection& Worker)
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp]:
```suggestion
bool WriteWorkerSection(TSharedRef<TJsonWriter<>> Writer, const FWorkerTypeLaunchSection& Worker)
```

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+				Writer->WriteValue(TEXT("duration"), Worker.LoginRateLimit.Duration);
+				Writer->WriteValue(TEXT("requests_per_duration"), Worker.LoginRateLimit.RequestsPerDuration);
+			Writer->WriteObjectEnd();
+		}
+	Writer->WriteObjectEnd();
+
+	return true;
+}
+
+bool WriteLoadbalancingSection(TSharedRef< TJsonWriter<> > Writer, const FName& WorkerType, const int32 Columns, const int32 Rows, const bool ManualWorkerConnectionOnly)
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp]:
```suggestion
bool WriteLoadbalancingSection(TSharedRef<TJsonWriter<>> Writer, const FName& WorkerType, const int32 Columns, const int32 Rows, const bool ManualWorkerConnectionOnly)
```

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+}
+
+}
+
+bool GenerateDefaultLaunchConfig(const FString& LaunchConfigPath)
+{
+	if (const USpatialGDKEditorSettings* SpatialGDKEditorSettings = GetDefault<USpatialGDKEditorSettings>())
+	{
+		FString Text;
+		TSharedRef< TJsonWriter<> > Writer = TJsonWriterFactory<>::Create(&Text);
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp]:
```suggestion
		TSharedRef<TJsonWriter<>> Writer = TJsonWriterFactory<>::Create(&Text);
```

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+#define LOCTEXT_NAMESPACE "SpatialGDKDefaultLaunchConfigGenerator"
+
+using namespace SpatialGDK;
+
+namespace
+{
+bool WriteFlagSection(TSharedRef<TJsonWriter<>> Writer, const FString& Key, const FString& Value)
+{
+	Writer->WriteObjectStart();
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp]: This class appears to be using tabs whereas I think we normally use spaces?

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+
+	return true;
+}
+
+bool WriteWorkerSection(TSharedRef< TJsonWriter<> > Writer, const FWorkerTypeLaunchSection& Worker)
+{
+	Writer->WriteObjectStart();
+		Writer->WriteValue(TEXT("worker_type"), *Worker.WorkerTypeName.ToString());
+		Writer->WriteArrayStart(TEXT("flags"));
+		for (const auto& Flag : Worker.Flags)
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp]: Nit: I think this loop should be indented to match what happens elsewhere

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+		}
+	Writer->WriteObjectEnd();
+
+	return true;
+}
+
+bool WriteLoadbalancingSection(TSharedRef< TJsonWriter<> > Writer, const FName& WorkerType, const int32 Columns, const int32 Rows, const bool ManualWorkerConnectionOnly)
+{
+	Writer->WriteObjectStart();
+	Writer->WriteValue(TEXT("layer"), *WorkerType.ToString());
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp]: Nit: Should be indented after WriteObjectStart to match the rest of the class

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+			ClientWorker.bLoginRateLimitEnabled = false;
+			WriteWorkerSection(Writer, ClientWorker);
+			Writer->WriteArrayEnd(); // Worker section end
+		Writer->WriteObjectEnd(); // End of json
+
+		Writer->Close();
+
+		if (!FFileHelper::SaveStringToFile(Text, *LaunchConfigPath))
+		{
+			UE_LOG(LogSpatialGDKDefaultLaunchConfigGenerator, Log, TEXT("Failed to write output file '%s'. It might be that the file is read-only."), *LaunchConfigPath);
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp]: Shouldn't this be an error?

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp (hunk 5 above)
Comment (aleximprobable) [reply SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp]: No

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp (hunk 6 above)
Comment (aleximprobable) [reply SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp]: Updated.

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp (hunk 7 above)
Comment (aleximprobable) [reply SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp]: Updated.

File: SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp (hunk 8 above)
Comment (aleximprobable) [reply SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp]: Changed to Error.

**Pull request merged**